### ISSUE TITLE
After Update to 3.11.3: No pull requests are opened on Github

### ISSUE THREAD

**Issue by Flohack74** [WeblateOrg/weblate#4125]
**Describe the bug**
After upgrading from Weblate 3.8 to 3.11 2 weeks ago it seems that no more pull requests were opened for changes, also all repositories show that they have newer local changes.

Manually pushing the Rebase or Push button does not help. There are no obvious errors logged anywhere. Can you tell me in which logs to look for those events? I can only see stuff like

```
[2020-07-03 03:30:20,262: INFO/ForkPoolWorker-3] ubports/morph-browser: updating repository
[2020-07-03 03:30:20,263: INFO/ForkPoolWorker-3] Lock 140349893836528 acquired on /opt/weblate3/data/vcs/ubports/morph-browser.lock
[2020-07-03 03:30:20,276: DEBUG/ForkPoolWorker-3] hub: exec hub log -n 1 --format=format:%H @{upstream} [retcode=0]
[2020-07-03 03:30:21,511: DEBUG/ForkPoolWorker-3] hub: exec hub remote prune origin [retcode=0]
[2020-07-03 03:30:21,527: DEBUG/ForkPoolWorker-3] hub: exec hub branch --list --remote origin/* [retcode=0]
[2020-07-03 03:30:22,961: DEBUG/ForkPoolWorker-3] hub: exec hub fetch origin [retcode=0]
[2020-07-03 03:30:22,962: INFO/ForkPoolWorker-3] ubports/morph-browser: update took 2.70 seconds
[2020-07-03 03:30:22,976: DEBUG/ForkPoolWorker-3] hub: exec hub log -n 1 --format=format:%H @{upstream} [retcode=0]
[2020-07-03 03:30:22,977: INFO/ForkPoolWorker-3] ubports/morph-browser: repository updated from 80dfadc4101e085ed6c1bbb13f21437a5d1b8630 to 80dfadc4101e085ed6c1bbb13f21437a5d1b8630
[2020-07-03 03:30:22,978: DEBUG/ForkPoolWorker-3] ubports/morph-browser: update: 80dfadc4101e085ed6c1bbb13f21437a5d1b8630
[2020-07-03 03:30:22,986: INFO/ForkPoolWorker-3] Lock 140349893836528 released on /opt/weblate3/data/vcs/ubports/morph-browser.lock
```

So it seems the hub tool is still available and being used on the Github repos. But no push actions can be seen. Any ideas?

**Server configuration and status**
```
 * Weblate: 3.11.3
 * Django: 2.2.13
 * siphashc: 1.3
 * Whoosh: 2.7.4
 * translate-toolkit: 2.5.1
 * lxml: 4.5.1
 * Pillow: 7.0.0
 * bleach: 3.1.5
 * six: 1.14.0
 * python-dateutil: 2.8.1
 * social-auth-core: 3.2.0
 * social-auth-app-django: 3.1.0
 * django-crispy-forms: 1.8.1
 * oauthlib: 3.1.0
 * django-compressor: 2.4
 * djangorestframework: 3.11.0
 * django-appconf: 1.0.4
 * user-agents: 2.1
 * filelock: 3.0.12
 * setuptools: 47.3.0
 * jellyfish: 0.7.2
 * openpyxl: 2.6.4
 * celery: 4.4.5
 * kombu: 4.6.10
 * celery-batches: 0.3
 * translation-finder: 1.8
 * html2text: 2020.1.16
 * pycairo: 1.19.1
 * pygobject: 3.36.1
 * diff-match-patch: 20181111
 * requests: 2.23.0
 * django-redis: 4.11.0
 * hiredis: 1.0.1
 * sentry_sdk: 0.14.4
 * Cython: 0.29.20
 * misaka: 2.1.1
 * GitPython: 3.1.3
 * borgbackup: 1.1.13
 * Python: 3.5.2
 * Git: 2.7.4
 * phply: 1.2.5
 * chardet: 3.0.4
 * ruamel.yaml: 0.16.10
 * hub: 2.7.4
 * Database backends: django.db.backends.mysql
 * Cache backends: avatar:FileBasedCache, default:MemcachedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: box.ubports.com
 * Celery: redis://localhost:6379, redis://localhost:6379, regular
 * Platform: Linux 4.4.0-184-generic (x86_64)

SystemCheckError: System check identified some issues:

ERRORS:
?: (weblate.E006) Please migrate your database to use PostgreSQL. Support for other database backends will be dropped in Weblate 4.0 currently sheduled on April 2020.
        HINT: https://docs.weblate.org/en/weblate-3.11.3/admin/install.html#production-database

WARNINGS:
?: (security.W021) You have not set the SECURE_HSTS_PRELOAD setting to True. Without this, your site cannot be submitted to the browser preload list.
?: (weblate.W025.ass) Failure in loading handler for ass file format: aeidon or gaupol package required for Subtitle support
        HINT: https://docs.weblate.org/en/weblate-3.11.3/admin/install.html#optional-deps
?: (weblate.W025.srt) Failure in loading handler for srt file format: aeidon or gaupol package required for Subtitle support
        HINT: https://docs.weblate.org/en/weblate-3.11.3/admin/install.html#optional-deps
?: (weblate.W025.ssa) Failure in loading handler for ssa file format: aeidon or gaupol package required for Subtitle support
        HINT: https://docs.weblate.org/en/weblate-3.11.3/admin/install.html#optional-deps
?: (weblate.W025.sub) Failure in loading handler for sub file format: aeidon or gaupol package required for Subtitle support
        HINT: https://docs.weblate.org/en/weblate-3.11.3/admin/install.html#optional-deps

INFOS:
?: (weblate.I021) Error collection is not configured, it is highly recommended for production use
        HINT: https://docs.weblate.org/en/weblate-3.11.3/admin/install.html#collecting-errors
?: (weblate.I028) Backups are not configured, it is highly recommended for production use
        HINT: https://docs.weblate.org/en/weblate-3.11.3/admin/backup.html
?: (weblate.I031) New Weblate version is available, please upgrade to 4.1.
        HINT: https://docs.weblate.org/en/weblate-3.11.3/admin/upgrade.html

```

**Comment by Flohack74** [comment]
Well its even worse, I checked our Gitlab stuff where we directly commit (no MRs) and also there no pushes are being made...

**Comment by Flohack74** [comment]
So we have a setup on Github where Weblate pushes into its forks in a separate user, then opens a pull request for the main repo. I can now see that the changes are pushed to the forked repos, but no PR is opened. How can I debug this further? I dont see any hub errors in the logs...

**Comment by nijel** [comment]
Check logs for `hub merge-request` result. The problem here is that it fails with error when there is existing open pull request open, so the code assumes this, but there might be other error as well.

The long term fix for this is https://github.com/WeblateOrg/weblate/issues/3493

**Comment by Flohack74** [comment]
Does not seem to have any merge-request in any of the celery logs. I also unpacked archived ones, but thats strange, it should have been in logs before already. But actually we are doing PRs via a crontab entry, so probably its no in the celery log, does that make sense?

**Comment by nijel** [comment]
It still should be in Weblate logs, see https://docs.weblate.org/en/latest/contributing/debugging.html#weblate-logs

**Comment by Flohack74** [comment]
The usual Weblate logs are in the log file of the virtualenv folder, but now I see that they are rotated too quickly for anything useful to be there. So to sum up, if I press "Push" button on a repo typed Github, I should see merge-request log entry in the normal log?

**Comment by nijel** [comment]
Yes, it should execute hub merge-request command.

**Comment by Flohack74** [comment]
So the only logging I get when I press the push button on one of those repos is that:
```
2020-07-07 06:59:33,857 DEBUG hub: exec hub log -n 1 --format=format:%H @{upstream} [retcode=0]
2020-07-07 06:59:44,953 DEBUG hub: exec hub log -n 1 --format=format:%H @{upstream} [retcode=0]
2020-07-07 06:59:45,287 DEBUG hub: exec hub log --format=format:%H ..origin/xenial -- [retcode=0]
2020-07-07 06:59:45,320 DEBUG hub: exec hub log --format=format:%H origin/xenial.. -- [retcode=0]
2020-07-07 06:59:45,354 DEBUG hub: exec hub log --format=format:%H ..origin/xenial -- [retcode=0]
2020-07-07 06:59:45,388 DEBUG hub: exec hub log --format=format:%H origin/xenial.. -- [retcode=0]
2020-07-07 06:59:45,420 DEBUG hub: exec hub log -n 1 --format=format:%H @{upstream} [retcode=0]
2020-07-07 06:59:45,461 DEBUG hub: exec hub status [retcode=0]
```

**Comment by nijel** [comment]
That looks like there are no new commits to push. Can you edit some string and try with that?

**Comment by Flohack74** [comment]
Well all my repos have now pending updateds like this:

```
On branch xenial
Your branch is ahead of 'origin/xenial' by 19 commits.
  (use "git push" to publish your local commits)
nothing to commit, working directory clean
```
Thats what I am saying, Weblate suddenly does not see the need anymore to open a PR. The data is pushed to the Weblate-owned repo, but the PR is not made against origin.

**Comment by nijel** [comment]
Having a log that would show the push would help to look into the issue. It works well on several installations I know, so it has to be some issue specific to your setup.

**Comment by Flohack74** [comment]
@nijel there is no more log than what I posted in https://github.com/WeblateOrg/weblate/issues/4125#issuecomment-654642459 ... Its really weird, this has worked now many years over all upgrades without problems.

**Comment by nijel** [comment]
To get better log, you need to create additional commit on the repo. That will trigger push and should trigger creating a pull request. I've added some extra logging in bda946e000 which might help to diagnose what went wrong in that case.

**Comment by Flohack74** [comment]
@nijel I am afraid that  bda946e was not the right commit pointer ;)

**Comment by nijel** [comment]
Oops, sorry, off by one. afcfbbf78d051cc569852a76e72d1ac06527cc9e should be the correct one.

**Comment by Flohack74** [comment]
Okay this did not bring more insights. We decided to change all affected repos to Git type, and commit with squash and rebase so that we get as few commits as possible. Anyway nobody bothered too often to approve the PRs so they were forgotten and we lost again translations for releases. I will consider next major version upgrade, but currently its too much of a task ;)

**Comment by Flohack74** [comment]
Sorry to bother you again with this, but we had troubles getting the permissions right on Github, as we have protected branches everywhere and so i tried for one repo to switch back to Github pushing, to see if it magically heals itself. Well it doesn´t but now finally I get a reasonable error, see here: https://pastebin.com/kedetTUE
- Somehow it seems it does not use hub anymore, did anythjing change here in 3.11?
- While I am preparing an upgrade to 4.x do you think I can repair this in some way to get this at least workiong for a few weeks? Thanks!

**Comment by Flohack74** [comment]
Ok closing again: hub tool was used but seems to have been outdated. I upgraded to 2.14.x to fix this, was 2.2 before. You might want to give a note somewhere about the min requirement, in case its not there (did not look).

**Comment by nijel** [comment]
We're going to stop using hub soon because it's proven to be hard to debug and not behaving as we would like to, see https://github.com/WeblateOrg/weblate/pull/4224
